File: problem set 2/Readability /Readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//function for number
int letters(string word);
int words(string word);
int sentences(string word);




int main(void)
{

    // Get input words from both players
    string text = get_string("Text: ");







    // Score both words
    float x = letters(text);
    float y = words(text);
    float z = sentences(text);


    // TODO: Print the winner





//printf("%f \n",x);
//printf("%f \n",y);
//printf("%f \n",z);

    float r = (x / y) * 100 ;
    float e = (z / y) * 100 ;
    float c = (0.0588 * r - 0.296 * e - 15.8);

    int u = round(c);

// find out about grades
    if (u < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (u > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", u);
    }



}






















// forletters
int letters(string word)
{


    int a = strlen(word);
    int b = 0;
    int n = 0;
    int p = 0;
    char g;



    for (int i = 0 ; i < a ; i++)
    {

        g = word[i];

        if ((g >= 65 &&  g <= 90) || (g >= 97 &&  g <= 122))
        {
            b += 1;

        }

    }



    return b;


}






// for words
int words(string word)
{



    int c = strlen(word);
    int t = 0;
    char g;
    int i = 0;


    do
    {

        g = word[i];

        if ((g >= 65 &&  g <= 90)  || (g == 45) || (g == 39) || (g >= 97 &&  g <= 122))
        {

            t += 1;


            do
            {

                i += 1;
                g = word[i];

            }
            while ((g >= 65 &&  g <= 90)  || (g == 45) || (g == 39) || (g >= 97 &&  g <= 122)); // find word between a-z


        }
        else
        {
            i += 1;
        }

    }
    while (i < c);




    return t;

}












// for sentences
int sentences(string word)
{




    int a = strlen(word);
    int t = 0;
    int r = 0;
    int q = 0;
    char g;





    for (int i = 0 ; i < a ; i++)
    {

        g = word[i];

        if ((g == 46) || (g == 33) || (g == 63))
        {
            t += 1 ;

        }

    }


    return t;


}

File: problem set 6 /readability /readability.py
from cs50 import get_float
from cs50 import get_int
from cs50 import get_string


# forletters
def letters(word):
    a = len(word)
    b = 0
    n = 0
    p = 0

    for i in range(a):
        g = ord(word[i])
        if g >= 65 and g <= 90 or g >= 97 and g <= 122:
            b += 1

    return b


##############################################################################################################


def words(word):
    c = len(word)
    n = 0

    for h in range(c):
        g = ord(word[h])
        if g == 32:
            n += 1
    return n + 1


##############################################################################################################


# for sentences
def sentences(word):
    a = len(word)
    t = 0
    r = 0
    q = 0

    for i in range(a):
        g = ord(word[i])
        if g == 46 or g == 33 or g == 63:
            t += 1

    return t


# @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@


# Get input words from both players
word = get_string("Text: ")


# Score both words
x = float(letters(word))
y = float(words(word))
z = float(sentences(word))


r = float((x / y) * 100)
e = float((z / y) * 100)
c = float((0.0588 * r - 0.296 * e - 15.8))


u = int(round(c))


# find out about grades
if u < 1:
    print("Before Grade 1")
elif u > 16:
    print("Grade 16+")
else:
    print("Grade", u)
